BREAKING_CHANGES.md

```diff
@@ -32,12 +32,21 @@
 *   The `BranchPrefixToTrim` configuration property has been removed. `RegularExpression` is now used to capture named groups instead.
     *   Default `RegularExpression` for feature branches is changed from `^features?[/-]` to `^features?[/-](?<BranchName>.+)` to support using `{BranchName}` out-of-the-box
     *   Default `RegularExpression` for unknown branches is changed from `.*` to `(?<BranchName>.*)` to support using `{BranchName}` out-of-the-box
+*   The `Mainline` mode and the related implementation has been removed completely. The new `TrunkBased` version strategy should be used instead.
 *   The branch related property `is-mainline` in the configuration system has been renamed to `is-main-branch`
 *   The versioning mode has been renamed to deployment mode and consists of following values:
     *   ManualDeployment (previously ContinuousDelivery)
     *   ContinuousDelivery (previously ContinuousDeployment)
     *   ContinuousDeployment (new)
-	
+*   On the configuration root level a new property with name `version-strategy` has been introduced with following values:
+    *   ConfigNext
+    *   MergeMessage
+    *   TaggedCommit
+    *   TrackReleaseBranches
+    *   VersionInBranchName
+    *   NonTrunkBased = ConfigNext | MergeMessage | TaggedCommit | TrackReleaseBranches | VersionInBranchName,
```

Comment:
I'm not a fan of `NonTrunkBased`. Can we find another word that has the opposite meaning and thus doesn't require the `Non` prefix?

Reply:
Of course we can change it. Actually it's just a abbreviation for `ConfigNext,MergeMessage,TaggedCommit,TrackReleaseBranches,VersionInBranchName `for better usability. Do you have something in mind?

Reply:
What about just calling it `Default`?

Reply:
In my opinion default is everything and nothing. I would say to delete it completely.

Reply:
done

Reply:
`Default` would be what GitVersion would use if `strategies` is missing in `GitVersion.yml` and would be the same as `NonTrunkBased`. I think that makes sense, no?

Reply:
No that is not true default is empty. Because we have implemented it in the way it is abwarts compatible by using the GitFlowConfigurationBuilder if no workspace is defined. I future all configuration should have a workspace definition.

Reply:
You're right about `Default` being wrong here. I still think we should have an enum value for it, I just don't like the `NonTrunkBased` name. How about `BranchBased`?

With `workspace`, I suppose you're meaning `workflow`? If so, I agree.

Reply:
Yes I mean workflow. You are right at the moment we have no workspace term in our domain. Okay that is fine with me to name it BranchBased. What about the confusion with TrunkBased on workflow and strategy level? I thought maybe we should name the workflow TrunkBased and the strategy Mainline.

Reply:
I think we should use either `TrunkBased` or `Mainline`, not both. Would using `workflow: TrunkBased` be different from `strategies: [TrunkBased]`? If so, in what way?

Reply:
> I think we should use either `TrunkBased` or `Mainline`, not both. Would using `workflow: TrunkBased` be different from `strategies: [TrunkBased]`? If so, in what way?

Actually I'm quite at the beginning to understand what a trunk-based workflow is and how it should be ideally behave. That means this is just a beginng of an jorney for me and I think the workflow will evolve when the feedback is comming back from the community. At this moment I think a TrunkBased workflow should use the TrunkBasedVersionStrategy and the ConfiguredNextVersionVersionStrategy. But this can be changed anytime in future. Please notice that I have changed the workflow TrunkBased/v1.yaml already.